lib/ansible/modules/files/file.py

```diff
@@ -58,8 +59,8 @@
     type: path
   recurse:
     description:
-    - Recursively set the specified file attributes on directory contents.
-    - This applies only to C(state=directory).
+    - Recursively act on directory contents.
+    - This applies only to C(state=directory) or C(state=absent).
```

Comment:
We might want to note the version this started applying to C(state=absent), since it looks like the option has been around since 1.1.

Reply:
This merge hasn't been updated in a year.  It's virtually impossible to fix the unsafe behavior due to the fact that so many tests depend on it.

Reply:
Yeah, a lot of things depend on it. The core team discussed an open issue for it on Friday https://github.com/ansible/ansible/issues/22248#issuecomment-774106501 without being aware that this PR existed, and consensus was that we'd like to try to move forward with fixing it, as early as possible for the next release cycle so we can get feedback.

Reply:
`ci_complete` can be added to the latest commit to ensure all tests run.